Description: fix AP properties update problem
Author: electricface <songwentai@deepin.com>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: https://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: 2019-11-14

--- network-manager-1.14.6.2.orig/src/devices/wifi/nm-wifi-ap.c
+++ network-manager-1.14.6.2/src/devices/wifi/nm-wifi-ap.c
@@ -794,9 +794,13 @@ nm_wifi_ap_update_from_properties (NMWif
 
 	g_object_freeze_notify (G_OBJECT (ap));
 
-	if (g_variant_lookup (properties, "Privacy", "b", &b) && b)
-		changed |= nm_wifi_ap_set_flags (ap, priv->flags | NM_802_11_AP_FLAGS_PRIVACY);
-
+	if (g_variant_lookup (properties, "Privacy", "b", &b)) {
+		if (b) {
+			changed |= nm_wifi_ap_set_flags (ap, priv->flags | NM_802_11_AP_FLAGS_PRIVACY);
+		} else {
+			changed |= nm_wifi_ap_set_flags (ap, priv->flags &~ NM_802_11_AP_FLAGS_PRIVACY);
+		}
+	}
 	v = g_variant_lookup_value (properties, "WPS", G_VARIANT_TYPE_VARDICT);
 	if (v) {
 		if (g_variant_lookup (v, "Type", "&s", &s)) {
@@ -872,13 +876,13 @@ nm_wifi_ap_update_from_properties (NMWif
 
 	v = g_variant_lookup_value (properties, "WPA", G_VARIANT_TYPE_VARDICT);
 	if (v) {
-		changed |= nm_wifi_ap_set_wpa_flags (ap, priv->wpa_flags | security_from_vardict (v));
+		changed |= nm_wifi_ap_set_wpa_flags (ap, security_from_vardict (v));
 		g_variant_unref (v);
 	}
 
 	v = g_variant_lookup_value (properties, "RSN", G_VARIANT_TYPE_VARDICT);
 	if (v) {
-		changed |= nm_wifi_ap_set_rsn_flags (ap, priv->rsn_flags | security_from_vardict (v));
+		changed |= nm_wifi_ap_set_rsn_flags (ap, security_from_vardict (v));
 		g_variant_unref (v);
 	}
 
